jdaviz/components/viewer_data_select.vue

```diff
@@ -142,7 +144,33 @@ module.exports = {
       uncertTrunc: this.uncertainty,
     }
   },
+  mounted() {
+    let element = document.getElementById(`target-${this.viewer.id}`).parentElement
+    while (element["tagName"] !== "BODY") {
+      if (["auto", "scroll"].includes(window.getComputedStyle(element).overflowY)) {
+        element.addEventListener("scroll", this.onScroll);
+      }
+      element = element.parentElement;
+    }
+  },
+  beforeDestroy() {
+    let element = document.getElementById(`target-${this.viewer.id}`).parentElement
+    while (element["tagName"] !== "BODY") {
+      if (["auto", "scroll"].includes(window.getComputedStyle(element).overflowY)) {
+        element.removeEventListener("scroll", this.onScroll);
+      }
+      element = element.parentElement;
+    }
+  },
   methods: {
+    onScroll(e) {
+      const dataMenuHeight = document.getElementById("menu-button").parentElement.getBoundingClientRect().height
```

Comment:
I'm surprised that this doesn't need viewer id as well.... is that just because we currently hardcode the heights of the menus (and therefore they're all the same)?

Reply:
Absolutely right, the use of a single ID across all viewers stems from their menus having a uniform hardcoded height of 42 px. Initially, I considered directly using the value, but opted for a slightly less coupled approach :)

Reply:
ok, technically there are still multiple elements with this same ID, and it probably just grabs the first one (right?), but I think that's fine for now since we keep the same height.  Might just be worth a comment in-line (either here or where the id is set) in case we ever have different heights per-menu.  Or if it's just as easy to append the viewer id in both places, then that is a little more future-proofed.